.github/workflows/python.yml

```diff
@@ -10,19 +10,18 @@ jobs:
     runs-on: ubuntu-latest
     steps:
     - uses: actions/checkout@v4
-    # NB: v1.4.0 covers Python 3.8.
-    - uses: ricardochaves/python-lint@v1.4.0
+    - uses: actions/setup-python@v5
       with:
-        python-root-list: lddtree.py pylint
-        use-pylint: true
-        use-pycodestyle: false
-        use-flake8: false
-        use-black: true
-        use-mypy: true
-        use-isort: true
-        extra-pylint-options: ""
-        extra-pycodestyle-options: ""
-        extra-flake8-options: ""
-        extra-black-options: ""
-        extra-mypy-options: ""
-        extra-isort-options: ""
+        python-version: '3.x'
+    - name: install dependencies
+      run: |
+        python -m pip install pylint mypy black 'isort[colors]'
+        # mypy dependencies for following imported types
+        python -m pip install argcomplete
+    - run: pylint --output-format colorized lddtree.py
+    - run: mypy lddtree.py
+      env:
+        TERM: xterm-color
+        MYPY_FORCE_COLOR: 1
+    - run: black --check --diff --color lddtree.py
```

Comment:
one thing i disliked about the existing check is that it stopped on the first failing command.  would rather see every checker run and accumulate non-zero at the end.

Reply:
Here's an example of doing this: https://github.com/eli-schwartz/pax-utils/actions/runs/7435810723/job/20231562901

In order to demonstrate that it actually does work as expected, I had to also introduce a pylint error -- as you can see, the first linting step failed, but the later ones still ran, and the overall job exited with a failure.